Maemo/stretch

spinal84 opened the pull request:
* Add systemd support
* Add cross-compiling support
* Fix warnings
* Minor fixes

spinal84 committed:
Add cross-compiling support

spinal84 committed:
Move dsme-thermal logic from init script to libexec

This is needed because we can't use parts of bash scripts in systemd
service files

spinal84 committed:
Add systemd support

spinal84 committed:
[systemd] Use systemctl commands for shutdown/reboot

spinal84 committed:
dsme-wdd-wd.c: set watchdog_path size to 16 bytes

According to my research 16 bytes is enough to both suppress gcc warning
and to be able to safely store up to 99 watchdogs

spinal84 committed:
Update debian/changelog (0.61.6)

spinal84 committed:
Fix warning: dsme wdd: Couldn't remove lockfile

DSME doesn't create a lockfile, if it's not running in daemon mode

spinal84 committed:
Minor fixes

MerlijnWajer commented on modules/runlevel.c:
```diff
@@ -81,8 +81,30 @@ static void shutdown(dsme_runlevel_t runlevel)
            runlevel == DSME_RUNLEVEL_REBOOT   ? "Reboot"   :
                                                 "Malf");
 
+  /* If we have systemd, use systemctl commands */
```
Should this also go in `#ifdef DSME_SYSTEMD_ENABLE` ?

spinal84 replied:
This was done for simplicity. And I tried to stick the decisions made by Mer developers to make our code closer to what they have.

If you don't like this part, I can rewrite it.

MerlijnWajer replied:
No, it's good to keep it close to what Mer project does. Maybe one day we can just use their mer version.

MerlijnWajer commented on debian/control:
```diff
@@ -11,6 +11,7 @@ Build-Depends: debhelper (>= 10),
  libdbus-1-dev (>= 1.0),
  libdbus-glib-1-dev,
  maemo-system-services-dev,
+ libsystemd-dev,
```
Is there a way to make this optional (will this somehow cause trouble in Devuan)?

spinal84 replied:
It doesn't add troubles to Devuan. I already tested this to work in ascii-devel.
Devuan has support for libsystemd-dev and many packages in Devuan are linked to libsystemd0.
The purpose of Devuan is to not use systemd as init system, but it still uses it's libraries.
Anyway, this is just a build-time dependency which doesn't affect the resulting package for Devuan in any way.

MerlijnWajer replied:
Thanks for clarifying.

parazyd replied:
FWIW, in Devuan we also have libraries that can replace libsystemd and keep API compatibility.

MerlijnWajer commented:
Looks good to me. @freemangordon - got any opinions?

spinal84 committed:
Add LSB headers to init scripts

spinal84 committed:
OOM protection fixes

Fix kernel warning:
oom_adj is deprecated, please use /proc/476/oom_score_adj instead.

Also oom_score_adj uses -1000 value instead of -17 for oom_adj.

spinal84 committed:
[systemd] Drop socket activation logic

It didn't work because dsme forks before running dsme-server.

To implement socket activation we have to rewrite dsme-wdd.c logic, but
I don't see any sense in implementing socket activation in DSME.

So, let's just drop it.

spinal84 committed:
Update debian/changelog (0.61.7)

spinal84 committed:
Move dsme-exec-helper to /usr/lib/dsme
